lineapy/graph_reader/artifact_collection.py

```diff
@@ -42,55 +46,60 @@ def __init__(
         self.db: RelationalLineaDB = db
 
         self.input_parameters = input_parameters
+
+        # ------ Validate inputs ------
```

Comment:
split the three tasks performed by the former function into separate sections.
The advantage of having validation before fetching from db here is that if validation fails, we never have to touch the db. communication with IO can always be assumed to be more expensive than simple code-execution.